### ISSUE TITLE
[a11y]: ai-chat should provide label for form feedback window

### ISSUE THREAD

**Issue by Daniel-Schulz-Private** [carbon-design-system/carbon-labs#866]
### Package version

0.18.0

### React version

18.3.1

### Automated testing tool and ruleset

n/a

### Assistive technology

JAWS

### Description

The form feedback window label and role are not announced - making it difficult to the users to understand its purpose of the modal window.

Recommended fix: Provide the modal with a clear and descriptive label that communicates its exact purpose. This can be achieved by providing a visual heading with H2, and use the Same as Modal Label by using `aria-labeledby` (currently, a div with copy "Tell us about your experience" and styles is used instead).

Impact for screen reader users: When a modal lacks a descriptive label, screen reader users may not understand the modal's purpose upon its activation. This creates confusion, increases cognitive load, and may prevent users from interacting with the modal content effectively instead)

<img width="620" height="264" alt="Image" src="https://github.com/user-attachments/assets/3f2f35ad-4da0-4be6-8799-08a0f8154d5f" />

### WCAG 2.1 Violation

WCAG Mapping: 2.4.6 Headings and Labels

### Reproduction/example

n/a

### Steps to reproduce

1. open form feedback for a response message in ai-chat
2. click thumbs up or down to initiate feedback
3. verify that the form feedback window does not announce label and role using JAWS

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-labs/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon-labs/issues) for duplicate problems
